### PR TITLE
bpo-35134: Add Include/cpython/pytime.h file

### PR DESCRIPTION
**Pull request by nw0**
Add Include/cpython/pytime.h header file.

Move CPython C API from Include/pytime.h into a new
Include/cpython/pytime.h header file, which is included by
Include/pytime.h.

https://bugs.python.org/issue35134

In the same vein as previous PRs by @vstinner. I've never done this before, so apologies in advance if I've messed anything up.

https://bugs.python.org/issue35134

### CODE REVIEW

**Comment by nw0** [comment]
Hmm, I'd like to add the skip-news label like previous PRs for this issue, but can't seem to do this. Help?

**Comment by vstinner** [comment]
Hum, since the new Include/pytime.h is basically empty. I suggest to leave the file unchanged and the move it into Include/cpython/.

I don't expect anyone to include it directly, and it's already included by "Python.h". The C API documentation asks to only include "Python.h."

File: Modules/gcmodule.c [hunk 1]
```diff
@@ -31,7 +31,7 @@
 #include "pycore_pyerrors.h"
 #include "pycore_pystate.h"     // _PyThreadState_GET()
 #include "pydtrace.h"
-#include "pytime.h"             // _PyTime_GetMonotonicClock()
```
Comment (vstinner) [inline Modules/gcmodule.c:34]: Hum. Since it's included by Python.h, including it explicitly is not needed, no?

File: Include/cpython/pytime.h [hunk 2]
```diff
@@ -2,8 +2,7 @@
 #ifndef Py_PYTIME_H
 #define Py_PYTIME_H
 
-#include "pyconfig.h" /* include for defines */
-#include "object.h"
+#include "../object.h"
```
Comment (vstinner) [inline Include/cpython/pytime.h]: I don't think that people should include pytime.h directly, and so I suggest to simply remove these includes. Let people respect the Python C API documentation which suggests to **only** include "Python.h".

File: Modules/gcmodule.c (hunk 1 above)
Comment (nw0) [reply Modules/gcmodule.c:34]: I've updated these, and moved the `pytime.h` include further down in Python.h to allow for it. I believe there aren't any other files including `pytime.h` directly now.

**Comment by vstinner** [comment]
Thank you, I merged your PR.

**Pull request merged**